onnxruntime/test/providers/vitisai/vitisai_basic_test.cc

```diff
@@ -0,0 +1,316 @@
+  
+  int uploadElapsed = getElapsedTime(start);
+
+  // copy GPU buffer to CPU (readback)
+  void* output = nullptr;
+
+  // calling GPU->CPU readback function and measuring time
+  start = getStartingTime();
+  // copy GPU buffer to CPU (readback)
+  output = vaip::tensor_proto_new_d3d12_gpu_to_cpu(bufferGPU, d3d12_device.Get(), cmdList.Get(),
+                                                   byteSize, dxQueue);
+
+  int readbackElapsed = getElapsedTime(start);
+
+  //// Write output to file
+  // std::ofstream fw("result_dst1d.txt", std::ofstream::out);
+  //// check if file was successfully opened for writing
+  // if (fw.is_open()) {
+  //   // store array contents to text file
+  //   for (int i = 0; i < size; i++) {
+  //         fw << static_cast<int*>(output)[i];
+  //         fw << "\n";
+  //   }
+  //   fw.close();
+  // }
+
+  free(output);
+
+  Struct s;
+  s.upload = uploadElapsed;
+  s.readback = readbackElapsed;
+  return s;
+}
+
+
+TEST(VitisAIExecutionProviderTest, VAIPTest) {
+
+  // add input rgb tensor wxhx4 int8
+  int8_t* array1 = new int8_t[3840 * 1260 * 4];
+  int8_t* array2 = new int8_t[2560 * 1440 * 4];
+  int8_t* array3 = new int8_t[1920 * 1080 * 4];
+
+  Struct res_3840x1260 = VAIExecutionProviderTest<int8_t>(array1, 3840, 1260, 4);
+  delete[] array1;
+  Struct res_2560x1440 = VAIExecutionProviderTest<int8_t>(array2, 2560, 1440, 4);
+  delete[] array2;
+  Struct res_1920x1080 = VAIExecutionProviderTest<int8_t>(array3, 1920, 1080, 4);
+  delete[] array3;
+
+  // 1D input data
+  int8_t* int8_10k = new int8_t[10000];
+  int8_t* int8_100k = new int8_t[100000];
+  int8_t* int8_1m = new int8_t[1000000];
+  int8_t* int8_10m = new int8_t[10000000];
+
+  // check if float is 32 bit
+  assert(CHAR_BIT * sizeof(float) == 32);
+
+  float* fp32_10k = new float[10000];
+  float* fp32_100k = new float[100000];
+  float* fp32_1m = new float[1000000];
+  float* fp32_10m = new float[10000000];
+
+   Struct res_fp32_10k = VAIExecutionProviderTest<float>(fp32_10k, 10000);
+   Struct res_int8_10k = VAIExecutionProviderTest<int8_t>(int8_10k, 10000);
+   Struct res_fp32_100k = VAIExecutionProviderTest<float>(fp32_100k, 100000);
+   Struct res_int8_100k = VAIExecutionProviderTest<int8_t>(int8_100k, 100000);
+   Struct res_fp32_1m = VAIExecutionProviderTest<float>(fp32_1m, 1000000);
+   Struct res_int8_1m = VAIExecutionProviderTest<int8_t>(int8_1m, 1000000);
+   Struct res_fp32_10m = VAIExecutionProviderTest<float>(fp32_10m, 10000000);
+   Struct res_int8_10m = VAIExecutionProviderTest<int8_t>(int8_10m, 10000000);
+
+   delete[] int8_10k;
+   delete[] int8_100k;
+   delete[] int8_1m;
+   delete[] int8_10m;
+   delete[] fp32_10k;
+   delete[] fp32_100k;
+   delete[] fp32_1m;
+   delete[] fp32_10m;
+
+  system("wmic path win32_VideoController get name > gpu_info.txt");
+  system("wmic cpu get name > sys_info.txt");
+  system("wmic memorychip get speed >> sys_info.txt");
```

Comment:
Please use `std::system` instead of the function from the global namespace. Please check the result and report failure if it it's occurred.